tests/schema/extensions/schema_extensions/conftest.py

```diff
@@ -0,0 +1,123 @@
+import contextlib
+import dataclasses
+import enum
+from typing import Any, AsyncGenerator, List, Type
+
+import pytest
+
+import strawberry
+from strawberry.extensions import SchemaExtension
+
+
+@dataclasses.dataclass
+class SchemaHelper:
+    query_type: type
+    subscription_type: type
+    query: str
+    subscription: str
+
+
+class ExampleExtension(SchemaExtension):
+    def __init_subclass__(cls, **kwargs: Any):
+        super().__init_subclass__(**kwargs)
+        cls.called_hooks = []
+
+    expected = [
+        "on_operation Entered",
+        "on_parse Entered",
+        "on_parse Exited",
+        "on_validate Entered",
+        "on_validate Exited",
+        "on_execute Entered",
+        "resolve",
+        "resolve",
+        "on_execute Exited",
+        "on_operation Exited",
+        "get_results",
+    ]
+    called_hooks: List[str]
+
+    @classmethod
+    def assert_expected(cls) -> None:
+        assert cls.called_hooks == cls.expected
+
+
+@pytest.fixture()
+def default_query_types_and_query() -> SchemaHelper:
+    @strawberry.type
+    class Person:
+        name: str = "Jess"
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def person(self) -> Person:
+            return Person()
+
+    @strawberry.type
+    class Subscription:
+        @strawberry.subscription
+        async def count(self) -> AsyncGenerator[int, None]:
+            for i in range(5):
+                yield i
+
+    subscription = "subscription TestSubscribe { count }"
+    query = "query TestQuery { person { name } }"
+    return SchemaHelper(
+        query_type=Query,
+        query=query,
+        subscription_type=Subscription,
+        subscription=subscription,
+    )
+
+
+class ExecType(enum.Enum):
+    SYNC = enum.auto()
+    ASYNC = enum.auto()
+
+    def is_async(self) -> bool:
+        return self == ExecType.ASYNC
+
+
+@pytest.fixture(params=[ExecType.ASYNC, ExecType.SYNC])
+def exec_type(request: pytest.FixtureRequest) -> ExecType:
+    return request.param
+
+
+@contextlib.contextmanager
+def hook_wrap(list_: List[str], hook_name: str):
+    list_.append(f"{hook_name} Entered")
+    try:
+        yield
+    finally:
+        list_.append(f"{hook_name} Exited")
```

Comment:
nit: maybe a "category" style would be better than a human readable one in this case?

no strong feelings though

```suggestion
    list_.append(f"enter: {hook_name}")
    try:
        yield
    finally:
        list_.append(f"exit: {hook_name}")
```

Reply:
This would change many tests. I would not do this because it would trigger file changed for who already reviewed them. LMK if its critical for you.